core/domain/classifier_services.py

```diff
@@ -356,3 +356,151 @@ def delete_classifier_training_job(job_id):
+        exp_version: int. The exploration version.
+        state_name: str. The state to which the classifier belongs.
+        classifier_id: str. ID of the classifier.
+
+    Returns:
+        mapping_id: str. ID of the classifier exploration mapping.
+    """
+    mapping_id = classifier_models.ClassifierExplorationMappingModel.create(
+        exp_id, exp_version, state_name, classifier_id)
+    return mapping_id
+
+def _update_classifier_exploration_mapping(mapping_model,
+                                           classifier_id):
+    """Updates classifier exploration mapping model in the datastore.
+
+    Args:
+        mapping_model: ClassifierExplorationMappingModel. Classifier exploration
+            mapping instance in datastore.
+        classifier_id: The ID of the classifier.
+
+    Note: All of the properties of a classifier exploration mapping are
+        immutable, except for classifier_id (classifier_id becomes mutable when
+        we implement forward propagation).
+    """
+    mapping_model.classifier_id = classifier_id
+    mapping_model.put()
+
+def save_classifier_exploration_mapping(exp_id, exp_version, state_name,
+                                        classifier_id):
+    """Checks for the existence of the model.
+    If the model exists, it is updated using
+        _update_classifier_exploration_mapping method.
+    If the model doesn't exist, it is created using
+        _create_classifier_exploration_mapping method.
+
+    Args:
+        exp_id: str. ID of the exploration.
+        exp_version: int. The exploration version.
+        state_name: str. The state to which the classifier belongs.
+        classifier_id: str. ID of the classifier.
+
+    Returns:
+        mapping_id : str. ID of the classifier exploration mapping.
+    """
+    mapping_id = exp_id + str(exp_version) + state_name
+    classifier_exploration_mapping_model = (
+        classifier_models.ClassifierExplorationMappingModel.get(mapping_id,
+                                                                False))
+    if classifier_exploration_mapping_model is None:
+        classifier_exploration_mapping = (
+            classifier_domain.ClassifierExplorationMapping(
+                mapping_id, classifier_id))
+        classifier_exploration_mapping.validate()
+        mapping_id = _create_classifier_exploration_mapping(exp_id, exp_version,
+                                                            state_name,
+                                                            classifier_id)
+    else:
+        classifier_exploration_mapping = (
+            get_classifier_exploration_mapping_from_model(
+                classifier_exploration_mapping_model))
+        classifier_exploration_mapping.validate()
+        _update_classifier_exploration_mapping(
+            classifier_exploration_mapping_model, classifier_id)
+    return mapping_id
+
+def delete_classifier_exploration_mapping(exp_id, exp_version, state_name):
```

Comment:
Nope, let's only implement functions when they are needed, otherwise we end up with dead code.

Why would you delete the old classifiers? Explorations can be reverted to previous versions; it might be useful to maintain that information.